src/main/scala/com/google/cloud/spark/bigquery/BigQueryWriteHelper.scala

```diff
@@ -0,0 +1,101 @@
+package com.google.cloud.spark.bigquery
+
+import java.util.UUID
+
+import collection.JavaConverters._
+import com.google.cloud.bigquery.{BigQuery, BigQueryException, FormatOptions, JobInfo, LoadJobConfiguration}
+import com.google.cloud.http.BaseHttpServiceException
+import com.typesafe.scalalogging.StrictLogging
+import org.apache.hadoop.fs.{Path, RemoteIterator}
+import org.apache.spark.sql.{DataFrame, SQLContext, SaveMode}
+
+case class BigQueryWriteHelper(bigQuery:BigQuery,
+                               sqlContext: SQLContext,
+                               saveMode: SaveMode,
+                               options: SparkBigQueryOptions,
+                               data: DataFrame) extends StrictLogging {
+
+  import org.apache.spark.sql.SaveMode
+
+  val conf = sqlContext.sparkSession.sparkContext.hadoopConfiguration
+
+  val temporaryGcsPath = {
+
+    val gcsPathOption = options.temporaryGcsBucket.map(bucket =>
+      s"gs://$bucket/.spark-bigquery-${System.currentTimeMillis()}-${UUID.randomUUID()}")
```

Comment:
nit: It's usually friendlier to use the spark appid or some such instead of currenttimemillis
You can find an id in SparkEnv.get

Reply:
Fixed in https://github.com/GoogleCloudPlatform/spark-bigquery-connector/pull/66/commits/9444852da1aa2884bbf922b04e10aa93a2008d4e